### PR TITLE
Upgrade Mobility Data Validator to 4.2.0

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

4.2.0 brings new validators! And a few changes which we account for here

### CODE REVIEW

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Great update thanks

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,9 @@ tmp/
 .tags
 *.pid
 
+# Ignore mobility data jar for building rules
+*.jar
```
Comment (philip-cline) [inline .gitignore:12]: love this.

File: lib/manager/components/validation/MobilityDataValidationResult.js [hunk 2]
```diff
@@ -139,7 +139,7 @@ const MobilityDataValidationResult = ({notice}) => {
             <p><Markdown>{rule.description || ''}</Markdown></p>
             <p>
               <a
-                href={`https://gtfs-validator.mobilitydata.org/rules.html#${notice.code}`}
+                href={`https://gtfs-validator.mobilitydata.org/rules.html#${notice.code}-rule`}
```
Comment (philip-cline) [inline lib/manager/components/validation/MobilityDataValidationResult.js:142]: clever!

**Review by philip-cline — APPROVED** [review]
Looks great to me!

**Pull request merged**